### DISCUSSION TITLE
when a pull request triggers a workflow, what code does it use?

### DISCUSSION THREAD

**Discussion by kidbrax** [community/community#25041] [Actions]
<p>Say I am meerging a feature branch into master so I create a pull request. This triggers my workflow. The workflow uses </p>
<p>actions/checkout. Does it use the code from the feature branch or what the code wold look like after merged?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>If your workflow is configured with on: pull_request, in default, checkout action get the merged code. </p>
<p>You could <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/managing-a-workflow-run#enabling-step-debug-logging" rel="nofollow noopener">enable step debug logging</a> and output GITHUB context to see the ref (like refs/pull/{ID}/merge) and you could find it in Checkout step logs. </p>
<p><img src="https://user-images.githubusercontent.com/24806478/181088537-5b4cd9d7-65c3-4cce-a243-bc2033069de8.png" alt="pr merge.png" data-base62-sha1="2UdMbuAXonhOjc1vHFh65Gh34pO" width="633" height="487"> </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/5350ca9f7f48fd2fbaecb22cad0477b52dd42822.png" data-download-href="/uploads/short-url/bT2IX0RGutbOjmSG5YOiIV9oKA2.png?dl=1" title="check out pr merge.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181088538-e8653e03-7af7-45c2-9d77-8ef9322ab111.png" alt="check out pr merge.png" data-base62-sha1="bT2IX0RGutbOjmSG5YOiIV9oKA2" width="690" height="470" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/5350ca9f7f48fd2fbaecb22cad0477b52dd42822_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">check out pr merge.png</span><span class="informations">775×528 18.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>There is my workflow example:</p>
<p><img src="https://user-images.githubusercontent.com/24806478/181088539-fba32f18-8df9-4df4-b6f3-7c7244bacfee.png" alt="checkout.png" data-base62-sha1="blqUBFUNOTmHiOOtyUXaDL7d3CT" width="371" height="280"></p>
